### ISSUE TITLE
Error loading two-sided quadratic inequalities

### ISSUE THREAD

**Issue by KSepetanc** [jump-dev/KNITRO.jl#167]
I am getting very long error while trying to load two-sided quadratic inequalities. It is not a big problem since I can load them as two separate constraints, but I see that it is supported by the solver so it is reasonable to support it with the link.

Used constraint:
`epsilon=1e-3`
`@constraint(sd, cosf_cce[t in t,(n,m) in buspairs;Ccond[(t,n,m)]], -epsilon/2 <= lambda14[t,(n,m)]*cv1[t,(n,m)] + lambda15[t,(n,m)]*cv2[t,(n,m)] + lambda16[t,(n,m)]*cv3[t,(n,m)] <= epsilon/2)`

```
LoadError: MathOptInterface.AddConstraintNotAllowed{MathOptInterface.ScalarQuadraticFunction{Float64},MathOptInterface.Interval{Float64}}: Adding `MathOptInterface.ScalarQuadraticFunction{Float64}`-in-`MathOptInterface.Interval{Float64}` constraints cannot be performed. You may want to use a `CachingOptimizer` in `AUTOMATIC` mode or you may need to call `reset_optimizer` before doing this operation if the `CachingOptimizer` is in `MANUAL` mode.
in expression starting at C:\Users\karlo\OneDrive\gmsdir\OPF\BL2_J\CC_eps.jl:269
correct_throw_add_constraint_error_fallback(::KNITRO.Optimizer, ::MathOptInterface.ScalarQuadraticFunction{Float64}, ::MathOptInterface.Interval{Float64}; error_if_supported::MathOptInterface.AddConstraintNotAllowed{MathOptInterface.ScalarQuadraticFunction{Float64},MathOptInterface.Interval{Float64}}) at constraints.jl:156
correct_throw_add_constraint_error_fallback(::KNITRO.Optimizer, ::MathOptInterface.ScalarQuadraticFunction{Float64}, ::MathOptInterface.Interval{Float64}) at constraints.jl:155
throw_add_constraint_error_fallback(::KNITRO.Optimizer, ::MathOptInterface.ScalarQuadraticFunction{Float64}, ::MathOptInterface.Interval{Float64}; kwargs::Base.Iterators.Pairs{Union{},Union{},Tuple{},NamedTuple{(),Tuple{}}}) at constraints.jl:129
throw_add_constraint_error_fallback(::KNITRO.Optimizer, ::MathOptInterface.ScalarQuadraticFunction{Float64}, ::MathOptInterface.Interval{Float64}) at constraints.jl:129
add_constraint(::KNITRO.Optimizer, ::MathOptInterface.ScalarQuadraticFunction{Float64}, ::MathOptInterface.Interval{Float64}) at constraints.jl:119
_broadcast_getindex_evalf at broadcast.jl:648 [inlined]
_broadcast_getindex at broadcast.jl:621 [inlined]
getindex at broadcast.jl:575 [inlined]
copy at broadcast.jl:876 [inlined]
materialize at broadcast.jl:837 [inlined]
add_constraints at constraints.jl:175 [inlined]
add_constraints(::MathOptInterface.Bridges.LazyBridgeOptimizer{KNITRO.Optimizer}, ::Array{MathOptInterface.ScalarQuadraticFunction{Float64},1}, ::Array{MathOptInterface.Interval{Float64},1}) at bridge_optimizer.jl:1079
copy_constraints(::MathOptInterface.Bridges.LazyBridgeOptimizer{KNITRO.Optimizer}, ::MathOptInterface.Utilities.UniversalFallback{MathOptInterface.Utilities.Model{Float64}}, ::MathOptInterface.Utilities.IndexMap, ::Array{MathOptInterface.ConstraintIndex{MathOptInterface.ScalarQuadraticFunction{Float64},MathOptInterface.Interval{Float64}},1}) at copy.jl:228
pass_constraints(::MathOptInterface.Bridges.LazyBridgeOptimizer{KNITRO.Optimizer}, ::MathOptInterface.Utilities.UniversalFallback{MathOptInterface.Utilities.Model{Float64}}, ::Bool, ::MathOptInterface.Utilities.IndexMap, ::Array{DataType,1}, ::Array{Array{MathOptInterface.ConstraintIndex{MathOptInterface.SingleVariable,MathOptInterface.GreaterThan{Float64}},1},1}, ::Array{DataType,1}, ::Array{Array{T,1} where T,1}, ::typeof(MathOptInterface.Utilities.copy_constraints), ::typeof(MathOptInterface.set)) at copy.jl:265
pass_constraints(::MathOptInterface.Bridges.LazyBridgeOptimizer{KNITRO.Optimizer}, ::MathOptInterface.Utilities.UniversalFallback{MathOptInterface.Utilities.Model{Float64}}, ::Bool, ::MathOptInterface.Utilities.IndexMap, ::Array{DataType,1}, ::Array{Array{MathOptInterface.ConstraintIndex{MathOptInterface.SingleVariable,MathOptInterface.GreaterThan{Float64}},1},1}, ::Array{DataType,1}, ::Array{Array{T,1} where T,1}) at copy.jl:240
default_copy_to(::MathOptInterface.Bridges.LazyBridgeOptimizer{KNITRO.Optimizer}, ::MathOptInterface.Utilities.UniversalFallback{MathOptInterface.Utilities.Model{Float64}}, ::Bool) at copy.jl:340
#automatic_copy_to#113 at copy.jl:15 [inlined]
automatic_copy_to at copy.jl:14 [inlined]
#copy_to#3 at bridge_optimizer.jl:268 [inlined]
(::MathOptInterface.var"#copy_to##kw")(::NamedTuple{(:copy_names,),Tuple{Bool}}, ::typeof(MathOptInterface.copy_to), ::MathOptInterface.Bridges.LazyBridgeOptimizer{KNITRO.Optimizer}, ::MathOptInterface.Utilities.UniversalFallback{MathOptInterface.Utilities.Model{Float64}}) at bridge_optimizer.jl:268
attach_optimizer(::MathOptInterface.Utilities.CachingOptimizer{MathOptInterface.AbstractOptimizer,MathOptInterface.Utilities.UniversalFallback{MathOptInterface.Utilities.Model{Float64}}}) at cachingoptimizer.jl:149
optimize!(::MathOptInterface.Utilities.CachingOptimizer{MathOptInterface.AbstractOptimizer,MathOptInterface.Utilities.UniversalFallback{MathOptInterface.Utilities.Model{Float64}}}) at cachingoptimizer.jl:185
optimize!(::Model, ::Nothing; bridge_constraints::Bool, ignore_optimize_hook::Bool, kwargs::Base.Iterators.Pairs{Union{},Union{},Tuple{},NamedTuple{(),Tuple{}}}) at optimizer_interface.jl:131
optimize! at optimizer_interface.jl:107 [inlined]
optimize!(::Model) at optimizer_interface.jl:107
top-level scope at CC_eps.jl:269
include_string(::Function, ::Module, ::String, ::String) at loading.jl:1088
```

**Comment by frapac** [comment]
Good catch. We should add `ScalarQuadraticFunction-in-IntervalSet` in the MOI wrapper.

**Comment by odow** [comment]
How is this even getting called? Why isn't this bridged? Do you have a full MWE?

**Comment by blegat** [comment]
If the wrapper just ends up splitting the constraint in a `ScalarQuadraticFunction`-in-`LessThan` and a `ScalarQuadraticFunction`-in-`GreaterThan` as KNITRO does not exploit the specific structure of a quadratic interval constraint then the MOI wrapper should **not** support it.
We should maybe update the error message saying

> If you are not using bridges, this means that the solver does not exploit the specific structure of this constraint but it does not mean that it cannot be transformed into constraints that the solver supports! Use `instantiate(..., with_bridge_type=Float64)` to apply a bridge layer that will try to bridge this constraint automatically.
> Otherwise, call `MOI.Bridges.debug_supports_constraint(...)` to see why it cannot be transformed to supported constraints.

**Comment by frapac** [comment]
Closed by #168
